### ISSUE TITLE
Misleading Document for conv1d and max1d in gluon

### ISSUE THREAD

**Issue by Godricly** [apache/incubator-mxnet#9704]
The document claims to support 'NWC' layout, while it actually not.
my current mxnet version is mxnet-cu80 (1.0.0.post4).

@zhreshold  @mli 
```
layout (str, default 'NCW') – Dimension ordering of data and weight. Can be ‘NCW’, ‘NWC’, etc. ‘N’, ‘C’, ‘W’ stands for batch, channel, and width (time) dimensions 
```

**Comment by gianlucacorrado** [comment]
Any reason why this issue was closed?

MXNet 1.0.0 and 1.1.0 return an MXNetError if I attempt at using layout='NWC':
`MXNetError: Invalid Input: 'NWC', valid values are: {None, 'NCDHW', 'NCHW', 'NCW', 'NDHWC', 'NHWC'}, in operator Convolution(name="", no_bias="False", num_group="1", num_filter="128", pad="(0,)", layout="NWC", dilate="(1,)", stride="(1,)", kernel="(5,)")`

Moreover, the list of valid values contains `None`, that id passed as value to layout generates a different error due to the fact that the code attempts at finding N, and C symbols.

`AttributeError: 'NoneType' object has no attribute 'find'`

**Comment by Godricly** [comment]
It was closed since no one had been watching.

**Comment by gianlucacorrado** [comment]
Can you reopen it since it wasn't fixed?

Thanks

**Comment by Godricly** [comment]
@gianlucacorrado reopened

**Comment by anjishnu** [comment]
Can we at least get the documentation fixed?

**Comment by Godricly** [comment]
@anjishnu is this one still there? I have not no idea where the doc is , otherwise I can submit a pr. @zhreshold. Hi Zhi, do you have time to check this, Please? Thank you.

**Comment by zhreshold** [comment]
I see, the doctoring for Conv1D(https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/nn/conv_layers.py#L198)
@Godricly Would you like to send a PR, thanks!

**Comment by Godricly** [comment]
Some updates here. Unlike the pooling layers which has an [assertion](https://github.com/apache/incubator-mxnet/blob/master/python/mxnet/gluon/nn/conv_layers.py#L719) to check the layout format, the conv layers didn't check it. So if I I used a 'NWC' layout,  the output error message is even more strange.  A possible solution is to add assertion into each conv1d, conv2d stuffs, and fix the corresponding documents.
@zhreshold @mli Any suggestion for this? 

Demo code:
```python
from mxnet import nd
from mxnet.gluon import nn

x = nd.ones((5,2,3))
conv = nn.Conv1D(layout='NWC', channels=4, kernel_size=3)
conv.initialize()
print(x.shape)
print(conv(x).shape)
```
Error Log:
```
mxnet.base.MXNetError: Invalid Input: 'NWC', valid values are: {None, 'NCDHW', 'NCHW', 'NCW', 'NDHWC', 'NHWC'}, in operator Convolution(name="", layout="NWC", pad="(0,)", num_filter="4", dilate="(1,)", num_group="1", stride="(1,)", no_bias="False", kernel="(3,)")
```

**Comment by Godricly** [comment]
fixed in #10915.
